### ISSUE TITLE
Standalone version?

### ISSUE THREAD

**Issue by joeldbirch** [Blocklevel/vue-social#1]
Hello!

I know this code is for use with your Blue framework, but would it be possible to adapt this to be a standalone version? I'd love to use it on the front-end of my Laravel project, but when I `import VueSocial from 'vue-social'` and run `npm run dev` (which just runs webpack) I get:

```bash
 ERROR  Failed to compile with 1 errors
This dependency was not found in node_modules:
* vue-social
```

I have inspected your code and while it looks fairly straight forward, I have only just begun to get my head around Vue.js so my skills are fairly n00bish.

If you don't plan on doing this, maybe you have a couple of tips for how to adapt this for my use case, please? Thank you for your time, regardless.

**Comment by MatteoGabriele** [comment]
Looks like you didn't install the plugin itself. Did you run `npm install vue-social`?

**Comment by joeldbirch** [comment]
Hey Matteo, thanks for the quick response! Yes, I did install the plugin and verified that the files were in my node_modules folder. I'm dearly hope I'm not that dense 😆

When I copy/paste the code from dist/@blocklevel/vue-social.js into the same folder as my main js file and reference it via `'./vue-social.js` rather than relying on the import finding the file from the installed package, the build compiles. However, I then get an error in the console.
```
Uncaught TypeError: Cannot set property 'auth' of undefined
```
If I manually define `Vue.social = {}` prior to the code that triggers the error, I then get the next error:
```
Uncaught TypeError: plugin.apply is not a function
    at Function.Vue.use
```
I know this isn't something you can discern remotely, but may I please get clarification that this package is supposed to work for my use case and not just for your framework? Thanks again!

**Comment by jonaskuiler** [comment]
Hey Joel,

Some questions:
- What Vue version are you using?
- What kind of setup are you using to build JavaScript, webpack?
- Can you send me a snippet of your main.js file?

Thanks

**Comment by jonaskuiler** [comment]
I think the problem is that you override the Vue.social object, you only have to configure it's members like `auth`.

I assume that if you override Vue.social it will override every function/property of the plugin.

**Comment by joeldbirch** [comment]
Hey, thanks for your reply. I really appreciate it! It's very late at night here (Australia) but I'll post more info when I get back to work. Cheers!

**Comment by joeldbirch** [comment]
Hello again. I have created a **[stripped down project showing the issue](https://github.com/joeldbirch/vue-social-issue)**. If you clone the repo and `npm install` then view index.html, you will see in the console:

> ```Uncaught TypeError: Cannot set property 'auth' of undefined```

### [>>Visit the stripped down test repo<<](https://github.com/joeldbirch/vue-social-issue)

The answers to Jona's questions are all answered by the test repo, but for convenience, here they are:

- **What Vue version are you using?**
Vue version 2.1.10
- **What kind of setup are you using to build JavaScript, webpack?**
The JS is being built with Webpack via Laravel Mix, which is just a standalone Webpack wrapper with presets.
- **Can you send me a snippet of your main.js file?**
Included in the repo, and below…

```js
window.Vue = require('vue');
window.VueSocial = require('@blocklevel/vue-social');

Vue.social.auth = {
  google: 'http://127.0.0.1:8000/auth/google'
}
```

Note, I had to use @blocklevel/vue-social to avoid issue #2.

**Comment by jonaskuiler** [comment]
Hey Joel,

Thanks for all the information! I really appreciate that :)
Because you're not using ES6 imports in your `main.js` file, when requiring the plugin via `require('@blocklevel/vue-social')` it will import an object with a default export inside, you can read more about ES6 import/exports here http://www.2ality.com/2014/09/es6-modules-final.html. 

Further more you're not adding the plugin to the Vue instance, Vue plugins use inversion of control to achieve a flexible architecture, you can read about that here https://vuejs.org/v2/guide/plugins.html.

So in a nutshell your `main.js` file should look like this: 

``` javascript
import Vue from 'vue'
import VueSocial from '@blocklevel/vue-social'

Vue.social.auth = {
  google: 'http://127.0.0.1:8000/auth/google'
}

Vue.use(VueSocial)
```

I've tested it in your repository and this works for me, let me know if you run into any problems.

**Comment by joeldbirch** [comment]
Hi again,

Thanks for looking at it!

I had originally used ES6 imports but avoided them for the test case in an effort to simplify things as much as possible. That's also why I didn't go further than the the `Vue.social.auth…` block.

Having updated my main.js file to what you showed above I'm not seeing any change at all, strangely. I have pushed the change to my test repo.

I wonder how this could possibly be working differently for us? Did you see the same error as me prior to using ES6 imports?

Thanks so much for persisting with this.
